pkg/segment/query/querystatus.go

```diff
@@ -176,18 +190,65 @@ func StartQuery(qid uint64, async bool, cleanupCallback func()) (*RunningQuerySt
 		return nil, fmt.Errorf("qid has already been started")
 	}
 
-	if len(allRunningQueries) >= MAX_RUNNING_QUERIES {
+	stateChan := make(chan *QueryStateChanData, numStates)
+
+	runningState := &RunningQueryState{
+		startTime:         time.Now(),
+		StateChan:         stateChan,
+		cleanupCallback:   cleanupCallback,
+		rqsLock:           &sync.Mutex{},
+		isAsync:           async,
+		timeoutCancelFunc: nil,
+	}
+
+	select {
+	case waitingQueries <- WaitStateData{qid, runningState}:
```

Comment:
Using a channel works for now. But soon we'll want to show how many queries are ahead of this one, and be able to remove a query from the queue if the user cancels the query. How do you plan on implementing that while using a channel here?

Reply:
Have changed the implementation to use slices with locks instead of channels. Could've used list but this implementation is much cleaner and copying a small slice (100 in this case) wouldn't affect performance.